zulip/integrations/matrix/matrix_bridge.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+import os
+import logging
+import signal
+import traceback
+
+from types import FrameType
+from typing import Any, Callable, Dict
+
+from matrix_bridge_config import config
+import zulip
+from matrix_client.client import MatrixClient
+
+def die(signal, frame):
+    # type: (int, FrameType) -> None
+
+    # We actually want to exit, so run os._exit (so as not to be caught and restarted)
+    os._exit(1)
+
+def zulip_to_matrix_username(full_name, site):
+    # type: (str, str) -> str
+    return "@**{0}**:{1}".format(full_name, site)
+
+def matrix_to_zulip(zulip_client, cfg):
+    # type: (Any, Dict[str, Any]) -> Callable[[Any, Dict[str, Any]], None]
+    def _matrix_to_zulip(room, event):
+        # type: (Any, Dict[str, Any]) -> None
+        """Matrix -> Zulip
+        """
+        if event['type'] == "m.room.member":
+            if event['membership'] == "join":
+                content = "{0} joined".format(event['content']['displayname'])
+        elif event['type'] == "m.room.message":
+            if event['content']['msgtype'] == "m.text":
+                content = "{0}: {1}".format(event['sender'], event['content']['body'])
+        else:
+            content = event['type']
+        msg_data = {
+            "sender": zulip_client.email,
+            "type": "stream",
+            "to": cfg["stream"],
+            "subject": cfg["subject"],
+            "content": content
+        }
+        not_from_bot = 'body' not in event['content'] or not event['content']['body'].startswith("@**")
```

Comment:
Can you explain this?

Reply:
`event['content']` doesn't always have the 'body' key.
1. When it doesn't contain the key, it **certainly** does not originate from the
   bot. Additionally its use is to prevent `KeyError`.
2. When it does, if the 'body' starts with the at-mention prefix that is
   specific to Zulip user's full name, it certainly does originate from the bot.